vizro-core/tests/unit/vizro/actions/_callback_mapping/test_get_action_callback_mapping.py

```diff
@@ -0,0 +1,267 @@
+                    value="gdpPercap",
+                ),
+            ),
+            vm.Parameter(
+                id="parameter_y",
+                targets=["scatter_chart.y", "scatter_chart_2.y"],
+                selector=vm.Dropdown(
+                    id="parameter_y_selector",
+                    options=["lifeExp", "gdpPercap", "pop"],
+                    multi=False,
+                    value="lifeExp",
+                ),
+            ),
+        ],
+    )
+    Vizro._pre_build()
+
+
+@pytest.fixture
+def export_data_inputs_expected():
+    return {
+        "filters": [
+            dash.State("filter_continent_selector", "value"),
+            dash.State("filter_country_selector", "value"),
+        ],
+        "parameters": [],
+        "filter_interaction": [
+            dash.State("scatter_chart", "clickData"),
+        ],
+        "theme_selector": [],
+    }
+
+
+@pytest.fixture
+def action_callback_inputs_expected():
+    return {
+        "filters": [
+            dash.State("filter_continent_selector", "value"),
+            dash.State("filter_country_selector", "value"),
+        ],
+        "parameters": [
+            dash.State("parameter_x_selector", "value"),
+            dash.State("parameter_y_selector", "value"),
+        ],
+        "filter_interaction": [
+            dash.State("scatter_chart", "clickData"),
+        ],
+        "theme_selector": dash.State("theme_selector", "on"),
+    }
+
+
+@pytest.fixture
+def action_callback_outputs_expected(request):
+    return {
+        f'{target.split(".")[0]}': dash.Output(target.split(".")[0], target.split(".")[1]) for target in request.param
+    }
+
+
+@pytest.fixture
+def export_data_outputs_expected(request):
+    return {
+        f"download-dataframe_{target}": dash.Output(
+            {"action_id": "export_data_action", "target_id": target, "type": "download-dataframe"}, "data"
+        )
+        for target in request.param
+    }
+
+
+@pytest.fixture
+def export_data_components_expected(request):
+    return [
+        dash.dcc.Download(id={"type": "download-dataframe", "action_id": "export_data_action", "target_id": target})
+        for target in request.param
+    ]
+
+
+@pytest.fixture
+def export_data_action_targets(request):
+    return request.param
+
+
+@pytest.mark.usefixtures("managers_one_page_four_controls_two_graphs_filter_interaction")
+class TestCallbackMapping:
+    """Tests action callback mapping for predefined and custom actions."""
+
+    @pytest.mark.parametrize(
+        "action_id, expected_inputs",
+        [
+            ("filter_action_filter_continent", "action_callback_inputs_expected"),
+            ("filter_interaction_action", "action_callback_inputs_expected"),
+            ("parameter_action_parameter_x", "action_callback_inputs_expected"),
+            ("on_page_load_action_action_test_page", "action_callback_inputs_expected"),
+            ("export_data_action", "export_data_inputs_expected"),
```

Comment:
`test_action_callback_mapping_inputs` tests input callback mappings for predefined actions. 

`_filter`, `filter_interaction`, `parameter`, and `_on_page_load` actions always have the same inputs (all the filters from the page, all the parameters from the page, filter_interactions and theme_selector) so the `expected_inputs` for those 4 actions is defined in the `action_callback_inputs_expected ` fixture. 

the `export_data` action is slightly different because the inputs are only filters from the screen and filter_interactions (export data isn't interested in selected parameters or theme_selector inputs). So, the `expected_input` for the `export_data` action is defined in "export_data_inputs_expected" fixture instead.